src/com/palmergames/bukkit/towny/command/PlotCommand.java

```diff
@@ -276,82 +277,29 @@ public boolean parsePlotCommand(Player player, String[] split) {
 						
 						if (selection.size() + resident.getTownBlocks().size()  > TownySettings.getMaxResidentPlots(resident))
 							throw new TownyException(Translation.of("msg_max_plot_own", TownySettings.getMaxResidentPlots(resident)));
-
-						double cost = 0;
-
-						// Remove any plots Not for sale (if not the mayor) and
-						// tally up costs.
-						for (WorldCoord worldCoord : new ArrayList<>(selection)) {
-							if (!worldCoord.hasTownBlock()) 
-								selection.remove(worldCoord);
-							else { 
-								TownBlock tb = worldCoord.getTownBlockOrNull();
-								if (tb == null) {
-									selection.remove(worldCoord);
-									continue;
-								}
-								double price = tb.getPlotPrice();
-								
-								if (tb.hasPlotObjectGroup()) {
-									// This block is part of a group, special tasks need to be done.
-									PlotGroup group = tb.getPlotObjectGroup();
-									
-									if (TownyEconomyHandler.isActive() && (!resident.getAccount().canPayFromHoldings(group.getPrice())))
-										throw new TownyException(Translation.of("msg_no_funds_claim_plot_group", group.getTownBlocks().size(), TownyEconomyHandler.getFormattedBalance(group.getPrice())));
-
-									// Add the confirmation for claiming a plot group.
-									Confirmation.runOnAccept(() -> {
-										ArrayList<WorldCoord> coords = new ArrayList<>();
-
-										// Get worldcoords from plot group.
-										group.getTownBlocks().forEach((tblock) -> coords.add(tblock.getWorldCoord()));
-
-										// Execute the plot claim.
-										new PlotClaim(Towny.getPlugin(), player, resident, coords, true, false, true).start();
-									})
-									.setTitle(Translation.of("msg_plot_group_claim_confirmation", group.getTownBlocks().size()) + " " + TownyEconomyHandler.getFormattedBalance(group.getPrice()) + ". " + Translation.of("are_you_sure_you_want_to_continue"))
+						
+						if (!resident.hasTown() && townBlock.getTownOrNull().isOpen() && !townBlock.getType().equals(TownBlockType.EMBASSY)) {
+							final Town town = townBlock.getTownOrNull();

```

Comment:
The max residents per town & max residents without nation settings seem to be ignored in this check.

Reply:
Good catch.